components/style/sharing/mod.rs

```diff
@@ -32,6 +34,88 @@ pub enum StyleSharingBehavior {
     Disallow,
 }
 
+/// Some data we want to avoid recomputing all the time while trying to share
+/// style.
+///
+/// This data structure is shared between the `ElementSharingChecker` and the
+/// `StyleSharingCandidate` (indeed we could try to merge those, though
+/// conceptually we may still want them to be separate).
+#[derive(Debug)]
+pub struct CachedStyleSharingData {
```

Comment:
I don't love "criteria", I don't think it says much, this is just a bunch of data we cache with the purpose of style sharing. But fine, I guess.